Orange/canvas/application/errorreporting.py

```diff
@@ -0,0 +1,207 @@
+                        continue
+                    v = '<a href="{}">{}</a>'.format(urljoin('file:', pathname2url(_v)), v)
+                if k == F.STACK_TRACE:
+                    v = v.replace('\n', '<br>').replace(' ', '&nbsp;')
+                lines.append('<tr><th align="left">{}:</th><td>{}</td></tr>'.format(k, v))
+            lines.append('</table>')
+            textbrowser.setHtml(''.join(lines))
+
+        settings = QSettings()
+        cb = QCheckBox(
+            'Include widget scheme (NOTE: your data sets will NOT '
+            'be transmitted)', self,
+            checked=settings.value('error-reporting/add-scheme', True, type=bool))
+        cb.stateChanged.connect(_reload_text)
+        _reload_text()
+
+        layout.addWidget(cb)
+        buttons = QWidget(self)
+        buttons_layout = QHBoxLayout(self)
+        buttons.setLayout(buttons_layout)
+        buttons_layout.addWidget(QPushButton('Submit', default=True, clicked=self.accept))
+        buttons_layout.addWidget(QPushButton('Dismiss', default=False, clicked=self.reject))
+        layout.addWidget(buttons)
+
+    def accept(self):
+        super().accept()
+        F = self.DataField
+        data = self._data.copy()
+
+        if QSettings().value('error-reporting/add-scheme', True, type=bool):
+            data[F.WIDGET_SCHEME] = data['_' + F.WIDGET_SCHEME]
+        del data['_' + F.WIDGET_SCHEME]
+
+        def _post_report(data):
+            for _retry in range(2):
+                try:
+                    urlopen(REPORT_POST_URL,
+                            timeout=10,
+                            data=urlencode(data).encode('utf8'))
+                except Exception as e:
+                    log.error('%s: %s', e.__class__.__name__, e.args[0])
+                    time.sleep(10)
+                    continue
+                break
+
+        Thread(target=_post_report, args=(data,)).start()
+
+        QMessageBox(QMessageBox.NoIcon, 'Thanks!',
+                    'Thanks for your bug report.<br><br>'
+                    'We are continually working '
+                    'hard to make our software more robust, faster, and '
+                    'easier to use. We also welcome contributions on '
+                    '<a href="{0}">{0}</a>.'.format(ISSUE_TRACKER_URL),
+                    QMessageBox.Ok, self,
+                    iconPixmap=QPixmap(THUMBSUP_IMAGE)).exec()
+
+    @classmethod
+    @patch('sys.excepthook', sys.__excepthook__)  # Prevent recursion
+    @pyqtSlot(object)
+    def handle_exception(cls, exc):
+        (etype, evalue, tb), canvas = exc
+
+        def _find_last_frame(tb):
+            while tb.tb_next:
+                tb = tb.tb_next
+            return tb
+
+        frame = _find_last_frame(tb)
+        cache_key = '{}:{}'.format(
+            frame.tb_frame.f_globals.get('__name__', frame.tb_frame.f_code.co_filename),
+            frame.tb_lineno)
+        if cache_key in cls._cache:
```

Comment:
I see the point in not showing the send error report dialog if an error occurs multiple times, but some kind of message notifying the user that something went wrong (again) would be nice.